app/src/Model/Table/ArticlesTable.php

```diff
@@ -0,0 +1,59 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Model\Table;
+
+use Cake\Event\EventInterface;
+use Cake\ORM\Table;
+use Cake\Utility\Text;
+use Cake\Validation\Validator;
+
+class ArticlesTable extends Table
+{
+    /**
+     * Initialize the table and add timestamp behavior.
+     *
+     * @param array $config Additional table configuration.
+     * @return void
+     */
+    public function initialize(array $config): void
+    {
+        $this->addBehavior('Timestamp');
+    }
+
+    /**
+     * Before saving an entity, generate a slug if it doesn't exist.
+     *
+     * @param \Cake\Event\EventInterface $event Save event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity being saved.
+     * @param array $options Additional options.
+     * @return void
+     */
+    public function beforeSave(EventInterface $event, $entity, $options)
+    {
+        if ($entity->isNew() && !$entity->get('slug')) {
+            $sluggedTitle = Text::slug($entity->get('title'));
+            // Trim the slug to the maximum length defined in the schema
+            $entity->set('slug', substr($sluggedTitle, 0, 191));
+        }
+    }
+
+    /**
+     * Define default validation rules for the articles entity.
+     *
+     * @param \Cake\Validation\Validator $validator Validator to be used.
+     * @return \Cake\Validation\Validator The configured validator.
+     */
+    public function validationDefault(Validator $validator): Validator
+    {
+        $validator
+            ->notEmptyString('title')
+            ->minLength('title', 10)
+            ->maxLength('title', 255)
+
+            ->notEmptyString('body')
+            ->minLength('body', 10);
+
+        return $validator;
```

Comment:
Field `slug` is unique, refer https://github.com/sergio9977/tutorial-cakephp/blob/d15d33cc99f3524041524d4e04595217dbafe18a/app/config/Migrations/20231031203043_Initial.php#L53-L61

How do we ensure the slug is unique?

Reply:
I added the title as unique